File: articles/organizational-change-institutional-adaptation/go/adaptive_change_score.go
// Adaptive Organizations: Institutional Change and Strategic Transformation
// Simple Go scoring demonstration.
//
// Responsible-use scope:
// Synthetic-data research, methods demonstration, institutional learning,
// and reproducible workflows only. Not for employment or individual worker decisions.

package main

import "fmt"

type AdaptiveChangeProfile struct {
	SensingQuality          float64
	InterpretiveCoherence  float64
	GovernanceAlignment    float64
	LearningCapability     float64
	ResourceFlexibility    float64
	InstitutionalLegitimacy float64
	OrganizationalInertia  float64
	Fragmentation          float64
	ResistanceFriction     float64
	EnvironmentalPressure  float64
}

func AdaptiveChangeCapacity(p AdaptiveChangeProfile) float64 {
	return 0.17*p.SensingQuality +
		0.16*p.InterpretiveCoherence +
		0.15*p.GovernanceAlignment +
		0.16*p.LearningCapability +
		0.12*p.ResourceFlexibility +
		0.12*p.InstitutionalLegitimacy -
		0.13*p.OrganizationalInertia -
		0.11*p.Fragmentation -
		0.10*p.ResistanceFriction -
		0.05*p.EnvironmentalPressure
}

func main() {
	profile := AdaptiveChangeProfile{84, 81, 79, 83, 74, 80, 20, 18, 24, 62}
	fmt.Printf("Synthetic adaptive change capacity score: %.2f\n", AdaptiveChangeCapacity(profile))
	fmt.Println("Responsible-use reminder: institutional learning only.")
}

File: articles/organizational-change-institutional-adaptation/cpp/adaptive_change_score.cpp
/*
Adaptive Organizations: Institutional Change and Strategic Transformation
Simple C++ scoring demonstration.

Responsible-use scope:
Synthetic-data research, methods demonstration, institutional learning,
and reproducible workflows only. Not for employment or individual worker decisions.
*/

#include <iostream>

struct AdaptiveChangeProfile {
    double sensing_quality;
    double interpretive_coherence;
    double governance_alignment;
    double learning_capability;
    double resource_flexibility;
    double institutional_legitimacy;
    double organizational_inertia;
    double fragmentation;
    double resistance_friction;
    double environmental_pressure;
};

double adaptive_change_capacity(const AdaptiveChangeProfile& p) {
    return
        0.17 * p.sensing_quality +
        0.16 * p.interpretive_coherence +
        0.15 * p.governance_alignment +
        0.16 * p.learning_capability +
        0.12 * p.resource_flexibility +
        0.12 * p.institutional_legitimacy -
        0.13 * p.organizational_inertia -
        0.11 * p.fragmentation -
        0.10 * p.resistance_friction -
        0.05 * p.environmental_pressure;
}

int main() {
    AdaptiveChangeProfile profile{84, 81, 79, 83, 74, 80, 20, 18, 24, 62};
    std::cout << "Synthetic adaptive change capacity score: "
              << adaptive_change_capacity(profile) << "\n";
    std::cout << "Responsible-use reminder: institutional learning only.\n";
    return 0;
}
